src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala

```diff
@@ -561,24 +561,34 @@ object DatabaseConversions extends LazyLogging {
         .flatMap {
           case (tag, updates) =>
             updates.map {
-              case OperationMetadata.BalanceUpdate(
-                  kind,
-                  change,
-                  category,
-                  contract,
-                  delegate,
-                  level
+              case BlockTagged(
+                  blockHash,
+                  blockLevel,
+                  _,
+                  cycle,
+                  period,
+                  OperationMetadata.BalanceUpdate(
+                    kind,
+                    change,
+                    category,
+                    contract,
+                    delegate,
+                    level
+                  )
                   ) =>
                 Tables.BalanceUpdatesRow(
                   id = 0,
                   source = tag.show,
                   sourceHash = hashing.get(from),
                   kind = kind,
-                  contract = contract.map(_.id),
+                  accountId = contract.map(_.id).orElse(delegate.map(_.value)).get, // either contract or delegate should be always set
```

Comment:
I suggest we program defensively, given that the balance updates is data coming from external sources that we can't control.
If both those values are missing, possibly for a breaking version update, the whole process will fail and stop Lorre until it's fixed.

We might choose to log an error if the value is missing and put a placeholder (empty string or "N/A"?) here.

It depends on how critical to business is the fact that this reference is absolutely filled correctly
@vishakh ?